core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts

```diff
@@ -0,0 +1,571 @@
+    let nodeDict2 = {
+      id: 'node_2',
+      thumbnail_filename: 'image.png',
+      title: 'Title 1',
+      description: 'Description 1',
+      prerequisite_skill_ids: ['skill_1'],
+      acquired_skill_ids: ['skill_2'],
+      destination_node_ids: ['node_3'],
+      outline: 'Outline',
+      exploration_id: null,
+      outline_is_finalized: false,
+      thumbnail_bg_color: '#a33f40',
+      status: 'Published',
+      planned_publication_date_msecs: 100,
+      last_modified_msecs: 100,
+      first_publication_date_msecs: 200,
+      unpublishing_reason: null
+    };
+    const learnerTopicSummaryBackendDict1 = {
+      id: 'sample_topic_id',
+      name: 'Topic Name',
+      language_code: 'en',
+      description: 'description',
+      version: 1,
+      story_titles: ['Story 1'],
+      total_published_node_count: 2,
+      thumbnail_filename: 'image.svg',
+      thumbnail_bg_color: '#C6DCDA',
+      classroom: 'math',
+      practice_tab_is_displayed: false,
+      canonical_story_summary_dict: [{
+        id: '0',
+        title: 'Story Title',
+        description: 'Story Description',
+        node_titles: ['Chapter 1'],
+        thumbnail_filename: 'image.svg',
+        thumbnail_bg_color: '#F8BF74',
+        story_is_published: true,
+        completed_node_titles: ['Chapter 1'],
+        url_fragment: 'story-title',
+        all_node_dicts: [nodeDict1]
+      },
+      {
+        id: '1',
+        title: 'Story Title',
+        description: 'Story Description',
+        node_titles: ['Chapter 1'],
+        thumbnail_filename: 'image.svg',
+        thumbnail_bg_color: '#F8BF74',
+        story_is_published: true,
+        completed_node_titles: [],
+        url_fragment: 'story-title',
+        all_node_dicts: [nodeDict2]
+      }],
+      url_fragment: 'topic-name',
+      subtopics: [subtopic],
+      degrees_of_mastery: {
+        skill_id_1: 0.5,
+        skill_id_2: 0.3
+      },
+      skill_descriptions: {
+        skill_id_1: 'Skill Description 1',
+        skill_id_2: 'Skill Description 2'
+      }
+    };
+    component.currentGoals = [LearnerTopicSummary.createFromBackendDict(
+      learnerTopicSummaryBackendDict1)];
+    component.goalTopics = [LearnerTopicSummary.createFromBackendDict(
+      learnerTopicSummaryBackendDict1)];
+    component.partiallyLearntTopicsList = [];
+    component.storyInProgress = [];
+    component.storyInRecommended = [];
+    component.incompleteExplorationsList = [];
+    component.explorationPlaylist = [];
+    component.collectionPlaylist = [];
+    component.completedExplorationsList = [];
+    component.untrackedTopics = {};
+    component.storyIdToLearnerGroupsTitleMap = new Map([
+      ['0', ['learner Group Title']]]);
+    component.username = 'username';
+    fixture.detectChanges();
+  });
+
+  it('should get the correct width in mobile view', () => {
+    expect(component.storyInProgress.length).toBe(1);
+  });
+
+  it('should check the completed ExpIds', () => {
```

Comment:
Not sure what this test does based on the description. Please rewrite it per https://github.com/oppia/oppia/wiki/Frontend-tests#name-tests-clearly

Reply:
Done PTAL